features/step_definitions/api-token.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const jwt = require('jsonwebtoken');
+const request = require('../support/request');
+
+module.exports = function server() {
+    this.Given(/^does not own a token named "([^"]*)"$/, tokenName =>
+        // Ensure there won't be a conflict: delete the token if it's already there
+        request({
+            uri: `${this.instance}/${this.namespace}/tokens`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            }
+        }).then((response) => {
+            const match = JSON.parse(response.body)
+                .find(token => token.name === tokenName);
+
+            if (!match) return Promise.resolve();
+
+            return request({
+                uri: `${this.instance}/${this.namespace}/tokens/${match.id}`,
+                method: 'DELETE',
+                auth: {
+                    bearer: this.jwt
+                }
+            });
+        }));
+
+    this.When(/^a new API token named "([^"]*)" is generated$/, tokenName =>
+        request({
+            uri: `${this.instance}/${this.namespace}/tokens`,
+            method: 'POST',
+            auth: {
+                bearer: this.jwt
+            },
+            body: {
+                name: tokenName
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 201);
+            Assert.strictEqual(response.body.lastUsed, '');
+            // Check that it's a base64 value of the right length to be a token
+            // encoded with https://www.npmjs.com/package/base64url
+            Assert.match(response.body.value, /[a-zA-Z0-9_-]{43}/);
+            this.userAccessToken = response.body.value;
+        }));
+
+    this.When(/^the token is used to log in$/, () =>
+        request({
+            uri: `${this.instance}/${this.namespace}/auth/logout`,
+            method: 'POST',
+            auth: {
+                bearer: this.jwt
+            }
+        }).then(() => this.getJwt(this.userAccessToken).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+
+            this.newJwt = response.body.token;
+
+            return null;
+        })));
+
+    this.Then(/^a valid JWT is received that represents "calvin"$/, () => {
```

Comment:
Oh, `authorization` also has one where the regexp has `"main"` in it, I guess this is probably fine then.